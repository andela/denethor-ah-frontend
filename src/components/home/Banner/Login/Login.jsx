import React from 'react';

export default () => <p>This is the Login Component</p>