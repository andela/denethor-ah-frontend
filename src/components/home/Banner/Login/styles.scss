@import '../../../../../src/styles/style.scss';

.social-login-block {
  display: flex;
  flex-direction: column;
  padding: 0 $m-size;
}

.social-login-block > p {
  margin: $s-size 0;
  color: white;
}

.login-block {
  display: flex;
  justify-content: center;
  padding: $m-size 0;
  font-family: Montserrat-Light;
}

.login-block__form {
  display: flex;
  flex-direction: column;
  padding: 0 $m-size;
  border-left: 1px solid whitesmoke;
}

.login-block__form {
  justify-content: flex-end;
}

.social-login-block-mobile {
  display: none
}

@media (max-width: 60rem) {
  .login-block__form {
    border: none;
  }

  .social-login-block-mobile {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: 0.5rem;
  }

  .social-login-block-mobile__caption {
    font-size: $s-size;
    color: white;
  }

  .social-login-block-mobile__buttons {
    display: flex;
    align-items: center;

    > button {
    height: 45px;
    width: 45px;
    margin: auto $s-size;
    border-radius: 50%;
    background-color: white;

      > svg {
        font-size: 1.25rem;
      }
    }
  }

  .social-login-block {
    display: none;
  }

  .login-block {
    flex-direction: column;
  }

  .login-block__form {
    border: none;
  }
}

.Toastify__toast--error.toast-custom-style {
  background: #fd2c27;
  border-radius: 0.2rem;
  font-family: Montserrat-Light;
  min-height: inherit;
  padding-right: 2rem;
  position: absolute;
  right: -2rem;
  > .Toastify__toast-body {
    font-size: 0.8rem;
  }
}

.spinner--button {
  border: 1px solid #f3f3f3; /* Light grey */
  border-top: 1px solid #3498db;
  display: block;
  margin: auto;
  width: 1rem;
  height: 1rem;
}

.hidden {
  display: none;
}