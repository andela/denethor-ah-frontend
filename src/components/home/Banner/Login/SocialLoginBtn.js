import React from 'react';

import './styles.scss';

const SocialLoginBtn = () => (
  <div className='social-login-block'>
    <p>Login with Social Media</p>
    <button className='button button--google'>Google</button>
    <button className='button button--facebook'>Facebook</button>
    <button className='button button--twitter'>Twitter</button>
  </div>
);

export default SocialLoginBtn;