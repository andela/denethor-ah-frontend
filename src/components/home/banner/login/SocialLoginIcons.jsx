import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGoogle, faFacebookF, faTwitter } from '@fortawesome/free-brands-svg-icons';

import './styles.scss';

const SocialLoginIcons = () => (
  <div className='social-login-block-mobile'>
    <div className='social-login-block-mobile__caption'>
      <p>Signup with or <b className='font-gold'>Login</b></p>
    </div>
    <div className='social-login-block-mobile__buttons'>
      <button><FontAwesomeIcon icon={faGoogle} color='#fd2c27' /></button>
      <button><FontAwesomeIcon icon={faFacebookF} color='#4267b2' /></button>
      <button><FontAwesomeIcon icon={faTwitter} color='#1da1f2' /></button>
    </div>
  </div>
);

export default SocialLoginIcons;