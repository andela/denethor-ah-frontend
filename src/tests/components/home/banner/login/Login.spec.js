import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../../../../components/home/banner/login/Login';

test('Should render login form', () => {
  const wrapper = shallow(<Login />);

  expect(wrapper).toMatchSnapshot();
});

it('Should set email state on input change', () => {
  const value = 'foobar';
  const wrapper = shallow(<Login />);
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('email')).toBe(value);
});

it('Should validate email on input blur', () => {
  const value = 'foobar';
  const wrapper = shallow(<Login />);
  wrapper.find('input').at(0).simulate('blur');
  expect(wrapper.state('emailErrorMessage')).toBe('Please enter your email address.');
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  wrapper.find('input').at(0).simulate('blur');
  expect(wrapper.state('emailErrorMessage')).toBe('Invalid email address, please crosscheck.');
  wrapper.find('input').at(0).simulate('change', {
    target: { value: 'lorem@ipsum.dolor' }
  });

  expect(wrapper.state('emailErrorMessage')).toBe('');
});

it('Should set password state on input change', () => {
  const value = 'foobar';
  const wrapper = shallow(<Login />);
  wrapper.find('input').at(1).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('password')).toBe(value);
});

it('Should validate password state on input blur', () => {
  const wrapper = shallow(<Login />);
  wrapper.find('input').at(1).simulate('blur');
  expect(wrapper.state('passwordErrorMessage')).toBe('Please enter your password.');

  wrapper.find('input').at(1).simulate('change', {
    target: { value: 'jgjgnf' }
  });
  wrapper.find('input').at(1).simulate('blur');
  expect(wrapper.state('passwordErrorMessage')).toBe('Password must be 8 or more characters.');

  wrapper.find('input').at(1).simulate('change', {
    target: { value: '$%^&*(*(*&^&*(' }
  });
  wrapper.find('input').at(1).simulate('blur');
  expect(wrapper.state('passwordErrorMessage')).toBe('Use numbers and letters for password.');

  wrapper.find('input').at(1).simulate('change', {
    target: { value: 'hgjvbkvgcfjvk7' }
  });
  wrapper.find('input').at(1).simulate('blur');
  expect(wrapper.state('passwordErrorMessage')).toBe('');
});

it('Should render error for form submission without email', () => {
  const wrapper = shallow(<Login />);
  wrapper.find('button').simulate('click', { preventDefault: () => {} });
  expect(wrapper.state('emailErrorMessage')).toBe('Please enter your email address.');
});

it('Should render error for form submission without password', () => {
  const value = 'foobar@gmail.com';
  const wrapper = shallow(<Login />);
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  wrapper.find('button').simulate('click', { preventDefault: () => {} });
  expect(wrapper.state('emailErrorMessage')).toBe('');
  expect(wrapper.state('passwordErrorMessage')).toBe('Please enter your password.');
});

it('Should call handleLogin for valid form submission', () => {
  const value = 'foobar@gmail.com';
  const handleLogin = jest.fn();
  const wrapper = shallow(<Login handleLogin={handleLogin} />);
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  wrapper.find('input').at(1).simulate('change', {
    target: { value }
  });
  wrapper.find('button').simulate('click', { preventDefault: () => {} });
  expect(wrapper.state('emailErrorMessage')).toBe('');
  expect(wrapper.state('passwordErrorMessage')).toBe('');
  expect(handleLogin).toHaveBeenCalled();
});
