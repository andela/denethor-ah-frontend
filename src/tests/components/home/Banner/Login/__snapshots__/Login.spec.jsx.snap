// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render login form 1`] = `
<div
  className="flex"
>
  <div
    className="login-block"
  >
    <SocialLoginBtn />
    <form
      className="login-block__form"
    >
      <input
        className=""
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="email"
        value=""
      />
      <input
        className=""
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
        value=""
      />
      <button
        className="button button--primary"
        onClick={[Function]}
      >
        <span
          className="button__text--login undefined"
        >
          Login
        </span>
        <span />
      </button>
    </form>
    <SocialLoginIcons />
  </div>
</div>
`;
